include/wx/clntdata.h

```diff
@@ -158,5 +159,38 @@ class WXDLLIMPEXP_BASE wxClientDataContainer
 
 };
 
+// This class is a replacement for wxClientDataContainer, and unlike
+// wxClientDataContainer the wxSharedClientDataContainer client data is
+// possible to copy (as a shared ptr) when instances of it are cloned.
+// Like wxClientDataContainer, wxSharedClientDataContainer is a mixin
+// that provides storage and management of "client data.". The client data
+// is reference counted and managed by the container.
+//
+// NOTE:  If your class has a clone function and needs to store client data,
+//        use wxSharedClientDataContainer and not wxClientDataContainer!
+
+class WXDLLIMPEXP_BASE wxSharedClientDataContainer
+{
+public:
+    void SetClientObject(wxClientData *data);
+    wxClientData *GetClientObject() const;
+    void SetClientData(void *data);
+    void *GetClientData() const;
+
+protected:
+    bool HasClientDataContainer() const {return NULL != m_data.get();}
+    wxSharedPtr<wxClientDataContainer> GetClientDataContainer() const {return m_data;}
+    void SetClientDataContainer(wxSharedPtr<wxClientDataContainer> data) {m_data = data;}
```

Comment:
Minor, but we don't use the Yoda comparison style and we also put spaces after/before braces, like this:
```suggestion
    bool HasClientDataContainer() const { return m_data.get() != NULL; }
    wxSharedPtr<wxClientDataContainer> GetClientDataContainer() const { return m_data; }
    void SetClientDataContainer(wxSharedPtr<wxClientDataContainer> data) { m_data = data; }
```